core/src/main/java/org/apache/carbondata/core/datamap/TableDataMap.java

```diff
@@ -165,4 +165,22 @@ public DataMapFactory getDataMapFactory() {
   @Override public void onEvent(Event event, OperationContext opContext) {
     dataMapFactory.fireEvent(event);
   }
+
+  /**
+   * Method to check whether the current segment need to be scanned or not for the queried data
+   *
+   * @param segmentId
+   * @param filterExp
+   * @return
+   * @throws IOException
+   */
+  public boolean isScanRequired(String segmentId, FilterResolverIntf filterExp) throws IOException {
```

Comment:
It should not be `isScanRequired`, it should be pruneSegments that should take segmentslist and filterexp, and returns the pruned list of segments